spot-vaults/contracts/_strategies/SpotCDRPricer.sol

```diff
@@ -0,0 +1,125 @@
+ *         We get the AMPL target price from Ampleforth's CPI oracle,
+ *         which is also used by the protocol to adjust AMPL supply through rebasing.
+ *
+ */
+contract SpotCDRPricer is ISpotPricingStrategy {
+    //-------------------------------------------------------------------------
+    // Libraries
+    using Math for uint256;
+
+    //-------------------------------------------------------------------------
+    // Constants & Immutables
+
+    uint256 private constant DECIMALS = 18;
+    uint256 private constant ONE = (10 ** DECIMALS);
+    uint256 public constant CL_ORACLE_DECIMALS = 8;
+    uint256 public constant CL_ORACLE_STALENESS_THRESHOLD_SEC = 3600 * 48; // 2 days
+    uint256 public constant USD_LOWER_BOUND = (99 * ONE) / 100; // 0.99$
+
+    /// @notice Address of the SPOT (perpetual tranche) ERC-20 token contract.
+    IPerpetualTranche public immutable SPOT;
+
+    /// @notice Address of the AMPL ERC-20 token contract.
+    IERC20 public immutable AMPL;
+
+    /// @notice Address of the USD token market price oracle.
+    IChainlinkOracle public immutable USD_ORACLE;
+
+    /// @notice Number of decimals representing the prices returned by the chainlink oracle.
+    uint256 public immutable USD_ORACLE_DECIMALS;
+
+    /// @notice Address of the Ampleforth CPI oracle. (provides the inflation-adjusted target price for AMPL).
+    IAmpleforthOracle public immutable AMPL_CPI_ORACLE;
+
+    /// @notice Number of decimals representing the prices returned by the ampleforth oracle.
+    uint256 public immutable AMPL_CPI_ORACLE_DECIMALS;
+
+    //-----------------------------------------------------------------------------
+    // Constructor
+
+    /// @notice Contract constructor.
+    /// @param spot Address of the SPOT token.
+    /// @param usdOracle Address of the USD token market price oracle token.
+    /// @param cpiOracle Address of the Ampleforth CPI oracle.
+    constructor(
+        IPerpetualTranche spot,
+        IChainlinkOracle usdOracle,
+        IAmpleforthOracle cpiOracle
+    ) {
+        SPOT = spot;
+        AMPL = IERC20(address(spot.underlying()));
+
+        USD_ORACLE = usdOracle;
+        USD_ORACLE_DECIMALS = usdOracle.decimals();
+
+        AMPL_CPI_ORACLE = cpiOracle;
+        AMPL_CPI_ORACLE_DECIMALS = cpiOracle.DECIMALS();
+    }
+
+    //--------------------------------------------------------------------------
+    // External methods
+
+    /// @return p The price of the usd token in dollars.
+    /// @return v True if the price is valid and can be used by downstream consumers.
+    function usdPrice() external view override returns (uint256, bool) {
+        (uint256 p, bool v) = _getCLOracleData(USD_ORACLE, USD_ORACLE_DECIMALS);
+        // If the market price of the USD coin fallen too much below 1$,
+        // it's an indication of some systemic issue with the USD token
+        // and thus its price should be considered unreliable.
+        return (ONE, (v && p > USD_LOWER_BOUND));
```

Comment:
WDYT about checking deviation USDC from $1.00, which also covers the case where the dollar token is >> $1? That's also a sign of market trouble, that's missed by just checking the downside.

Reply:
I don't think that's specifically a risk when it comes to USDC. If usdc is trading above one, folks will just mint more? Custody/Bank failure risk is only on the downside.

Reply:
I'm more coming from the perspective of-- IF the price of USDC is high, then something is going wrong.

Perhaps minting is halted by Circle corp (there are humans in the loop there), a liquidity pool address gets blacklisted through their AML system, there's an chainlink oracle attack/failure. Maybe other things as well I can't think of as well.

Reply:
Sure. Worth noting, that the current spot appraiser which is deployed and used by Billy doesn't have this check.